SimPEG/electromagnetics/static/resistivity/receivers.py

```diff
@@ -77,17 +73,45 @@ def eval(self, src, mesh, f):
         proj_f = self.projField
         if proj_f == "phi":
             proj_f = "phiSolution"
-        return P * f[src, proj_f]
+        v = P * f[src, proj_f]
 
-    def evalDeriv(self, src, mesh, f, v, adjoint=False):
+        if self.data_type == "apparent_resistivity":
+            try:
+                if mesh.dim == 2:
+                    return v / self.geometric_factor[src][:, None]
+                return v / self.geometric_factor[src]
+            except KeyError:
+                raise KeyError(
+                    "Receiver geometric factor has not been set, please execute "
+                    "survey.set_geometric_factor()"
+                )
+        return v
+
+    def evalDeriv(self, src, mesh, f, v=None, adjoint=False):
```

Comment:
so, this is called many times. 
can we encode `geometric_factor` in `P`? (I guess this was the previous implementation).

Reply:
Unfortunately not, which was the reason for switching. The geometric factor is dependent on the src, so either we store a `P` for every [source, receiver] , or a `geometric_factor` for every [source, receiver] pair. This is more memory efficient with only a slight hit to computation time. This also allows for a single receiver object to be attached to many sources.

Reply:
got it! that's a good point that I did not take into account before.